nerdlandbot/commands/pets.py

```diff
@@ -0,0 +1,309 @@
+                        name.capitalize(), pet_id
+                    )
+                    await ctx.send(msg)
+                else:
+                    msg = translate("pet_reaction_error", lang)
+                    await ctx.send(msg)
+
+            except asyncio.TimeoutError:
+                await guild_data.delete_pet(pet_id)
+                timeout = translate("command_timeout", lang)
+                embed = discord.Embed(
+                    title=embed_title,
+                    description=timeout,
+                )
+                return await ctx.send(embed=embed)
+
+    @commands.command(
+        name="post_pet",
+        aliases=["pet", "random_pet"],
+        brief="post_pet_brief",
+        help="post_pet_help",
+        usage="post_pet_usage",
+    )
+    @commands.guild_only()
+    async def post_pet(self, ctx: commands.Context, name: typing.Optional[str] = None):
+        lang, guild_data = await asyncio.gather(
+            culture(ctx),
+            get_guild_data(ctx.message.guild.id),
+        )
+
+        pets = guild_data.pets
+
+        if name == None:
+            pet_id = random.choice(list(pets.keys()))
+            guild_path = Path.cwd() / PETS_DIR_NAME / str(ctx.guild.id)
+            filepath = guild_path / f"{pet_id}.jpg"
+            message = translate("pet_posted", lang).format(
+                pet_id, pets[pet_id]["pet_name"].capitalize(), pets[pet_id]["category"].capitalize()
+            )
+        elif name in guild_data.pets_categories:
+            list_of_ids = list(pets.keys())
+            random.shuffle(list_of_ids)
+            for pet_id in list_of_ids:
+                if pets[pet_id]["category"] == name.lower():
+                    guild_path = Path.cwd() / PETS_DIR_NAME / str(ctx.guild.id)
+                    filepath = guild_path / f"{pet_id}.jpg"
+                    message = translate("pet_posted", lang).format(
+                        pet_id, pets[pet_id]["pet_name"].capitalize(), name.capitalize()
+                    )
+                    break
+            else:
+                message = translate("pet_not_found", lang)
+                return await ctx.send(message)
+        else:
+            list_of_ids = list(pets.keys())
+            random.shuffle(list_of_ids)
+            for pet_id in list_of_ids:
+                if pets[pet_id]["pet_name"] == name.lower():
+                    guild_path = Path.cwd() / PETS_DIR_NAME / str(ctx.guild.id)
+                    filepath = guild_path / f"{pet_id}.jpg"
+                    message = translate("pet_posted", lang).format(
+                        pet_id, pets[pet_id]["pet_name"].capitalize(), pets[pet_id]["category"].capitalize()
+                    )
+                    break
+            else:
+                message = translate("pet_not_found", lang)
+                return await ctx.send(message)
+
+        await ctx.send(message, file=discord.File(filepath))
```

Comment:
```suggestion
name = name.lower()
list_of_ids = list(pets.keys())
random.shuffle(list_of_ids)

def matching_first(pet: dict) -> bool:
	return True
	
def matching_category_name(pet: dict) -> bool:
	return pet["category"] == name
	
def matching_pet_name(pet: dict) -> bool:
    return pet["pet_name"] == name

# This can be formatted differently or written out in multiline
# if else statements, doesn't matter.
is_category_name = name in guild_data.pets_categories
matcher = (
	matching_first if name is None else (
		matching_category_name if is_category_name else matching_pet_name
	)
)
	

found_pet = next(
	pet for pet in pets.values() if matcher(pet=pet),
	None
)
if found_pet is None:
	message = translate("pet_not_found", lang)
    return await ctx.send(message)
    
# perhaps replace with get_pet_path() as mentioned before
guild_path = Path.cwd() / PETS_DIR_NAME / str(ctx.guild.id)
filepath = guild_path / f"{pet_id}.jpg"
message = translate("pet_posted", lang).format(
    pet["id],
    pet["pet_name"].capitalize(),
    pet["category"].capitalize(),
)
await ctx.send(message, file=discord.File(filepath))
```

Reply:
I tried this and it said that a generator expression should be parenthesized. Since that was parenthesized, I'm choosing to keep it the way it is.

Reply:
It's referring to adding parentheses here most likely:

```python
-pet for pet in pets.values() if matcher(pet=pet)
```